### PR TITLE
adds weeks to intervalToDuration

### PR DESCRIPTION
**Pull request by LucasHFS**
Related to #2612. This PR makes able to add the option to get the interval considering weeks in the calculation.

### CODE REVIEW

File: src/intervalToDuration/index.ts [hunk 1]
```diff
@@ -47,35 +48,73 @@ export default function intervalToDuration({ start, end }: Interval): Duration {
...
   }
 
-  const duration = {
-    years: 0,
-    months: 0,
-    days: 0,
-    hours: 0,
-    minutes: 0,
-    seconds: 0,
+  if(withWeeks){
```
Comment (wenderjean) [inline src/intervalToDuration/index.ts:51]: `if (...) {` :nail_care:

File: src/intervalToDuration/index.ts [hunk 2]
```diff
@@ -47,35 +48,73 @@ export default function intervalToDuration({ start, end }: Interval): Duration {
...
+    const remainingMinutes = sub(remainingHours, { hours: sign * duration.hours })
+    duration.minutes = Math.abs(differenceInMinutes(remainingMinutes, dateRight))
+  
+    const remainingSeconds = sub(remainingMinutes, {
+      minutes: sign * duration.minutes,
+    })
+    duration.seconds = Math.abs(differenceInSeconds(remainingSeconds, dateRight))
+  
+    return duration
+  }else{
```
Comment (wenderjean) [inline src/intervalToDuration/index.ts:87]: `} else {` :nail_care:

File: src/intervalToDuration/test.ts [hunk 3]
```diff
@@ -4,113 +4,165 @@ import assert from 'power-assert'
...
-    const result = intervalToDuration({ start, end })
-
-    assert.deepEqual(result, {
-      years: 39,
-      months: 2,
-      days: 20,
-      hours: 7,
-      minutes: 5,
-      seconds: 0,
+  describe('when withWeeks is false', function(){
```
Comment (wenderjean) [inline src/intervalToDuration/test.ts:7]: Looking for the left version it seems to be a common practice to put a space in the function declaration.
`function () {`

File: src/intervalToDuration/test.ts [hunk 4]
```diff
@@ -4,113 +4,165 @@ import assert from 'power-assert'
...
-        {
-          years: 0,
-          months: 0,
-          days: 1,
-          hours: 1,
-          minutes: 0,
-          seconds: 0,
-        }
-      )
+  // describe('edge cases', function () {
```
Comment (wenderjean) [inline src/intervalToDuration/test.ts:104]: Is this intentional?

Comment (LucasHFS) [reply src/intervalToDuration/test.ts]: nope

File: src/intervalToDuration/index.ts [hunk 5]
```diff
@@ -34,7 +35,7 @@ import sub from '../sub/index'
  * // => { years: 39, months: 2, days: 20, hours: 7, minutes: 5, seconds: 0 }
  */
 
-export default function intervalToDuration({ start, end }: Interval): Duration {
+export default function intervalToDuration({ start, end, withWeeks = false }: Interval): Duration {
```
Comment (leshakoss) [inline src/intervalToDuration/index.ts:38]: https://github.com/date-fns/date-fns/pull/2618#issuecomment-940415069

**Review by leshakoss — CHANGES_REQUESTED** [review]
